File: 0163-missing-ranges/0163-missing-ranges.cpp
class Solution {
public:
    vector<vector<int>> findMissingRanges(vector<int>& nums, int lower, int upper) {
        if (nums.size() == 0) return {{lower, upper}};
        vector<vector<int>> ans;
        if (nums[0] - lower >= 1) ans.push_back({lower, nums[0] - 1});
        for (int i = 0; i < nums.size() - 1; ++i){
            if (nums[i + 1] - nums[i] > 1) ans.push_back({nums[i] + 1, nums[i + 1] - 1});
        }
        if (upper - nums.back() >= 1) ans.push_back({nums.back() + 1, upper});
        return ans;
    }
};

File: 0163-missing-ranges/0163-missing-ranges.py
class Solution:
    def findMissingRanges(self, nums: List[int], lower: int, upper: int) -> List[List[int]]:
        if not nums:
            return [[lower, upper]]
        ans, i = [], 0
        
        if nums[0] - lower >= 1:
            ans.append([lower, nums[0] - 1])
        for i in range(len(nums) - 1):
            if nums[i + 1] - nums[i] > 1:
                ans.append([nums[i] + 1, nums[i + 1] - 1])
        if upper - nums[-1] >= 1:
            ans.append([nums[-1] + 1, upper])
                
        return ans